app/controllers/bulk_submission_excel/downloads_controller.rb

```diff
@@ -33,4 +35,17 @@ def create # rubocop:todo Metrics/AbcSize, Metrics/MethodLength
   def submission_parameters
     params.require(:bulk_submission_excel_download).permit(:asset_barcodes)
   end
+
+  private
+
+  # Build a filename for the file to be downloaded
+  # Consists of the template name in kebab-case, followed by the current date and time in ISO8601 format
+  # e.g. "submission-template_20190101T120000.xlsx"
+  def build_filename
+    template_name = params.dig(:defaults, :template_name) || 'Submission Template'
+    # convert to kebab-case, catching existing hyphens
+    template_name = template_name.downcase.gsub(/[-\s]+/, '-')
```

Comment:
This changes the existing behaviour when a plate is submitted. The first (tube) barcode could be used instead. User might expect seeing a barcode in the filename and they would be most likely to be unique.

However, "Limber-Htp - scRNA Core cDNA Prep" submission is done by the user in the story. The method makes a name like "limber-htp-scrna-core-cdna-prep_20190101T120000.xlsx", which may be indeed useful.

Reply:
Why don't we ask the users? You could post in the psd-techadmin channel (which contains the SSRs, who are the users) - present the two options and ask which they'd prefer?

I think I personally prefer Stephen's new format, but I see Abdullah's point - we would be changing the user experience for all submission templates this way.

Another option is a hybrid - keep the submission name in there, but use the first labware barcode instead of / as well as a timestamp?